exporter/awsprometheusremotewriteexporter/factory.go

```diff
@@ -0,0 +1,134 @@
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	prw "go.opentelemetry.io/collector/exporter/prometheusremotewriteexporter"
+)
+
+const (
+	typeStr = "awsprometheusremotewrite" // The value of "type" key in configuration.
+)
+
+// NewFactory returns a factory of the AWS Prometheus Remote Write exporter that can be registered to the Collector.
+func NewFactory() component.ExporterFactory {
+	return exporterhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporterhelper.WithMetrics(createMetricsExporter))
+}
+
+func createMetricsExporter(_ context.Context, params component.ExporterCreateParams,
+	cfg configmodels.Exporter) (component.MetricsExporter, error) {
+	// check if the configuration is valid
+	prwCfg, ok := cfg.(*Config)
+	if !ok {
+		return nil, errors.New("invalid configuration")
+	}
+
+	err := validateAndFixConfig(prwCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	client, cerr := prwCfg.HTTPClientSettings.ToClient()
+	if cerr != nil {
+		return nil, cerr
+	}
+
+	// load AWS auth configurations and create interceptor based on configuration
+	if applyAuth(prwCfg.AuthSettings) {
+		roundTripper, err := NewAuth(prwCfg.AuthSettings, client)
+		if err != nil {
+			return nil, err
+		}
+		client.Transport = roundTripper
+	}
+
+	// initialize an upstream exporter and pass it an http.Client with interceptor
+	prwe, err := prw.NewPrwExporter(prwCfg.Namespace, prwCfg.HTTPClientSettings.Endpoint, client, prwCfg.ExternalLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	// use upstream helper package to return an exporter that implements the required interface, and has timeout,
+	// queueing and retry feature enabled
+	prwexp, err := exporterhelper.NewMetricsExporter(
+		cfg,
+		params.Logger,
+		prwe.PushMetrics,
+		exporterhelper.WithTimeout(prwCfg.TimeoutSettings),
+		exporterhelper.WithQueue(prwCfg.QueueSettings),
+		exporterhelper.WithRetry(prwCfg.RetrySettings),
+		exporterhelper.WithShutdown(prwe.Shutdown),
+	)
+
+	return prwexp, err
+}
+
+func createDefaultConfig() configmodels.Exporter {
+	qs := exporterhelper.CreateDefaultQueueSettings()
+	qs.Enabled = false
+
+	ts := exporterhelper.CreateDefaultRetrySettings()
+	ts.Enabled = false
+
+	return &Config{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+		Namespace:       "",
+		TimeoutSettings: exporterhelper.CreateDefaultTimeoutSettings(),
+		RetrySettings:   ts,
+		QueueSettings:   qs,
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			Endpoint: "http://some.url:9411/api/prom/push",
+			// We almost read 0 bytes, so no need to tune ReadBufferSize.
+			ReadBufferSize:  0,
+			WriteBufferSize: 512 * 1024,
+			Timeout:         exporterhelper.CreateDefaultTimeoutSettings().Timeout,
+			Headers:         map[string]string{},
+		},
+	}
+}
+
+func validateAndFixConfig(prwCfg *Config) error {
+	err := prw.ValidateAndSanitizeExternalLabels(prwCfg)
```

Comment:
This specific line is what requires the referenced PR in the description to be merged